src/import-export/test/test-flat-bayes.c

```diff
@@ -0,0 +1,289 @@
+{
+    KvpFrame * frame;
+    Account * acc;
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    acc = gnc_imap_find_account_bayes (fixture->map1, fixture->tokens1);
+    g_assert_true (acc == fixture->account1);
+}
+
+static void
+find_flat_bayes_account (Fixture * fixture, gconstpointer pData)
+{
+    KvpFrame * frame;
+    Account * acc;
+    kvp_frame_set_string (qof_instance_get_slots (QOF_INSTANCE (fixture->book)),
+            "features/" GNC_FEATURE_GUID_FLAT_BAYESIAN, "value");
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account2);
+    acc = gnc_imap_find_account_bayes (fixture->map1, fixture->tokens1);
+    g_assert_true (acc == fixture->account1);
+}
+
+static void
+find_guid_bayes_account (Fixture * fixture, gconstpointer pData)
+{
+    KvpFrame * frame;
+    Account * acc;
+    kvp_frame_set_string (qof_instance_get_slots (QOF_INSTANCE (fixture->book)),
+            "features/" GNC_FEATURE_GUID_BAYESIAN, "value");
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account1);
+    gnc_imap_add_account_bayes (fixture->map1, fixture->tokens1, fixture->account2);
+    acc = gnc_imap_find_account_bayes (fixture->map1, fixture->tokens1);
+    g_assert_true (acc == fixture->account1);
+}
+
+static void
+find_normal_bayes_when_non_exists (Fixture * fixture, gconstpointer pData)
+{
+    KvpFrame * frame;
+    Account * acc;
+    acc = gnc_imap_find_account_bayes (fixture->map1, fixture->tokens1);
+    g_assert_true (acc == NULL);
+}
+
+static gchar const * suitename = "/import-export/test-flat-bayes";
+
+int
+main (int argc, char *argv[])
+{
+    int result;
+    qof_init();
+    g_test_init (&argc, &argv, NULL);
+    GNC_TEST_ADD (suitename, "normal_bayes_entry", Fixture, NULL, setup, &write_normal_bayes_entry, teardown);
+    GNC_TEST_ADD (suitename, "guid_bayes_entry", Fixture, NULL, setup, &write_guid_bayes_entry, teardown);
+    GNC_TEST_ADD (suitename, "normal_bayes_find", Fixture, NULL, setup, &find_normal_bayes_account, teardown);
+    GNC_TEST_ADD (suitename, "guid_bayes_find", Fixture, NULL, setup, &find_guid_bayes_account, teardown);
+    GNC_TEST_ADD (suitename, "guid_bayes_find", Fixture, NULL, setup, &find_normal_bayes_when_non_exists, teardown);
```

Comment:
This results in the following test failure:
```
GLib-ERROR **: duplicate test case path: /import-export/test-flat-bayes/guid_bayes_find
```

Can simply be solved by renaming one of these.